### PR TITLE
add confirm dialog before sharing files

### LINKED ISSUE #995: Add confirmation screen when uploading
There should be a in between screen when uploading a file so you can see you didn't misclick on a wrong file.
Bonus points for a cutting tool when it's an Image like in many other messengers.

But at least a yes/no to make sure the user knows there will not be an editing option

### PR DESCRIPTION
**Pull request by mahibi**
fix #995 
Signed-off-by: Marcel Hibbe <dev@mhibbe.de>

### CODE REVIEW

File: app/src/main/java/com/nextcloud/talk/controllers/ChatController.kt [hunk 1]
```diff
@@ -577,28 +579,46 @@ class ChatController(args: Bundle) : BaseController(args), MessagesListAdapter
         }
     }
 
-    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+    override fun onActivityResult(requestCode: Int, resultCode: Int, intent: Intent?) {
         if (requestCode == REQUEST_CODE_CHOOSE_FILE) {
             if (resultCode == RESULT_OK) {
-                uploadFile(data)
+                checkNotNull(intent)
```
Comment (tobiasKaminsky) [inline app/src/main/java/com/nextcloud/talk/controllers/ChatController.kt:585]: Is the resulting IllegalStateException somewhere handled?

File: app/src/main/java/com/nextcloud/talk/controllers/ChatController.kt [hunk 2]
```diff
@@ -577,28 +579,46 @@ class ChatController(args: Bundle) : BaseController(args), MessagesListAdapter
...
+
+                var filenamesWithLinebreaks = "\n"
+                files.forEach {
+                    var filename = UriUtils.getFileName(Uri.parse(it), context)
+                    filenamesWithLinebreaks += filename + "\n"
+                }
+
+                LovelyStandardDialog(activity)
+                        .setPositiveButtonColorRes(R.color.nc_darkGreen)
+                        .setTitle(context?.resources?.getString(R.string.nc_upload_confirm_send)?.let { String.format(it,
```
Comment (tobiasKaminsky) [inline app/src/main/java/com/nextcloud/talk/controllers/ChatController.kt:606]: This uses "send these files to" as plural also with one file.

**Review by tobiasKaminsky — APPROVED** [review]
Ready to merge, except one (very very) minor thing

File: app/src/main/java/com/nextcloud/talk/controllers/ChatController.kt (hunk 1 above)
Comment (mahibi) [reply app/src/main/java/com/nextcloud/talk/controllers/ChatController.kt]: done

File: app/src/main/java/com/nextcloud/talk/controllers/ChatController.kt (hunk 2 above)
Comment (mahibi) [reply app/src/main/java/com/nextcloud/talk/controllers/ChatController.kt]: done

**Comment by nickvergessen** [comment]
Screenshots of features like this are always nice to have in PRs :)

**Pull request merged**